Source/SpectrumAnalysis/KTChannelAggregator.hh

```diff
@@ -0,0 +1,145 @@
+/*
+ * KTChannelAggregator.hh
+ *
+ *  Created on: Jan 25, 2019
+ *      Author: P. T. Surukuchi
+ */
+
+#ifndef KTCHANNELAGGREGATOR_HH_
+#define KTCHANNELAGGREGATOR_HH_
+
+#include "KTProcessor.hh"
+#include "KTData.hh"
+#include "KTPowerSpectrumData.hh"
+#include "KTChannelAggregatedData.hh"
+#include "KTFrequencySpectrumDataFFTW.hh"
+#include "KTAxisProperties.hh"
+
+#include "KTSlot.hh"
+
+#include "KTMath.hh"
+
+namespace Katydid
+{
+  
+  KTLOGGER(avlog_hh, "KTChannelAggregator.hh");
+  
+  class KTChannelAggregatedData;
+  class KTPowerSpectrumData;
+  
+  /*
+   @class KTChannelAggregator
+   @author P. T. Surukuchi
+   
+   @brief Multiple channel summation for Phase-III and IV
+   
+   @details
+   More details to come.
+   
+   Configuration name: "channel-aggregator"
+   
+   Slots:
+   - "fft": void (Nymph::KTDataPtr) -- Adds channels voltages using FFTW-phase information for appropriate phase addition; Requires KTFrequencySpectrumDataFFTW; Adds summation of the channel results; Emits signal "fft"
+   
+   Signals:
+   - "fft": void (Nymph::KTDataPtr) -- Emitted upon summation of all channels; Guarantees KTFrequencySpectrumDataFFTW
+   */
+  
+  class KTChannelAggregator : public Nymph::KTProcessor
+  {
+  public:
+    KTChannelAggregator(const std::string& name = "channel-aggregator");
+    virtual ~KTChannelAggregator();
+    
+    bool Configure(const scarab::param_node* node);
+    
+    /*
+     Set radius of the active area assuming a circular active area.
+     */
+    virtual void SetActiveRadius(double);
+    
+    /*
+     Get radius of the active area.
+     */
+    virtual double GetActiveRadius() const;
+    
+    bool GetIsGridDefined() const;
+  private:
+    
+    // in meters, should not be hard-coded
+    // PTS: Has to come as an input from config file ?
+    double fActiveRadius = 0.0516;
+    
+    // Get the grid size assuming a square grid
+    // Set default to 30 currently
+    int fNGrid = 30;
+    
+    // PTS:: for 18.6 keV electrons, this somehow needs to come from the data file or config file
+    double fWavelength=0.0115;
+    
+    //For exception handling to make sure the grid is defined before the spectra are assigned.
+    bool fIsGridDefined=false;
+    
```

Comment:
Name variables in function definitions to make argument meanings clear from the header alone

Reply:
Not sure I understand the statement. Do you mean for me to change the argument in setter function for fIsGridDefined ? 
In commit 77127bb I used MEMBERVARIABLE() for fIsGridDefined. So, that might fix the issue ?